test/hooks.test.js

```diff
@@ -1,25 +1,34 @@
 'use strict'
 
-const tap = require('tap')
+const { test } = require('node:test')
 const { sink, once } = require('./helper')
 const pino = require('../')
 
-tap.test('log method hook', t => {
-  t.test('gets invoked', async t => {
+function match (t, obj, expected) {
+  const checkObj = Object.keys(expected).reduce((acc, key) => {
+    acc[key] = obj[key]
+    return acc
+  }, {})
+
+  t.assert.deepStrictEqual(checkObj, expected)
+}
```

Comment:
It's not available in v20 yet, but I would prefer we use https://github.com/nodejs/node/pull/54630. In the interim, we can copy https://github.com/newrelic/node-newrelic/blob/ebf3aa946be1e6802b567f2bfeef0b0c13613375/test/lib/custom-assertions/match.js into a file like `test/assert-match.js`. I'm sure it will be used in more suites than this one.

Reply:
@jsumners done. There are other PR that need this `match` assertion. I'll wait for this one to be merged before refactoring them.